### PR TITLE
Add support for writing multiple xml items in xml template expression

### LINKED ISSUE #26340: Error in how an `xml-template-expr` is handled with a sequence
**Description:**
$title.

**Steps to reproduce:**
```ballerina
import ballerina/io;

public function main() {
    xml x1 = xml `<foo>foo</foo><!--bar-->baz`;
    xml x2 = xml `<foo>foo</foo>`;
    io:println(x1);
    io:println(x1);
    io:println(x1 == x2);
}
```

This currently prints
```cmd
<foo>foo</foo>
<foo>foo</foo>
true
```

**Affected Versions:**
slp4

### PR DESCRIPTION
**Pull request by suleka96**
Fixes #26340

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/XmlSequence.java [hunk 1]
```diff
@@ -288,12 +289,26 @@ public void setChildren(BXml seq) {
      */
     @Override
     @Deprecated
-    public void addChildren(BXml seq) {
-        if (children.size() != 1) {
-            throw ErrorCreator.createError(StringUtils.fromString(("not an " + XmlNodeType.ELEMENT)));
+    public void addChildren(BXml xmlItem) {
```
[the remark this answers is not in the document]
Comment (suleka96) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/XmlSequence.java:292]: yes, it is only called during initialization, we can't call this function using any xml type or sub type.

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ValueCreator.java [hunk 2]
```diff
@@ -493,11 +492,7 @@ public static BXmlSequence createXmlSequence() {
      * @return xml sequence
      */
     public static BXmlSequence createXmlSequence(BArray sequence) {
-        List<BXml> children = new ArrayList<>();
-        for (Object value : sequence.getValues()) {
-            children.add((BXml) value);
-        }
-        return new XmlSequence(children);
+        return XmlFactory.createXmlSequence(sequence);
```
[the remark this answers is not in the document]
Comment (suleka96) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ValueCreator.java:495]: Seems like `public static BXmlSequence createXmlSequence(BArray sequence)` is not used anywhere atm.

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ValueCreator.java [hunk 3]
```diff
@@ -517,7 +512,7 @@ public static BXmlSequence createXmlSequence(List<BXml> sequence) {
      * @return xml sequence
      */
     public static BXmlSequence createXmlSequence(BXml child) {
-        return new XmlSequence(child);
+        return XmlFactory.createXmlSequence(child);
```
[the remark this answers is not in the document]
Comment (suleka96) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ValueCreator.java:515]: This is used in `xml filter` function. There are existing tests for that.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmInstructionGen.java [hunk 4]
```diff
@@ -1970,6 +1978,9 @@ void generateInstructions(int localVarOffset, BIRInstruction inst) {
                 case NEW_STRING_XML_QNAME:
                     generateNewStringXMLQNameIns((BIRNonTerminator.NewStringXMLQName) inst);
                     break;
+                case NEW_XML_SEQUENCE:
+                    generateXMLSequenceIns((BIRNonTerminator.NewXMLSequence) inst);
```
Comment (rdhananjaya) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmInstructionGen.java:1982]: generateNewXMLSequenceIns

**Comment by rdhananjaya** [comment]
We need to increment the BIR version https://github.com/ballerina-platform/ballerina-lang/blob/1c72b30aa0f6d873cb59b77bd7f6161b3a51927c/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/programfile/ProgramFileConstants.java#L30

And bir.ksy model (and add few tests cases) (I think this can be done in a separate PR too, if we are going to do this with a separate PR please create an issue)

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java [hunk 5]
```diff
@@ -4052,22 +4063,67 @@ public BLangNode transform(MatchStatementNode matchStatementNode) {
...
-                return createXMLLiteral(expressionNode);
+                return (BLangExpression) createXMLTextLiteral(xmlTypeNode);
+        }
+    }
+
+    public BLangNode createXmlTemplateLiteral(TemplateExpressionNode expressionNode) {
+        if (expressionNode.content().isEmpty()) {
+            return createXMLEmptyLiteral(expressionNode);
+        }
+        if (expressionNode.content().size() > 1) {
```
Comment (rdhananjaya) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java:4123]: I think it's more logical to order this, emtpy -> size == 1 -> size more than one

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java [hunk 6]
```diff
@@ -4052,22 +4063,67 @@ public BLangNode transform(MatchStatementNode matchStatementNode) {
         return matchStatement;
     }
 
-    public BLangNode createXmlTemplateLiteral(TemplateExpressionNode expressionNode) {
-        SyntaxKind contentKind;
-        if (expressionNode.content().isEmpty()) {
-            contentKind = SyntaxKind.XML_TEXT;
-        } else {
-            contentKind = expressionNode.content().get(0).kind();
+    public BLangXMLSequenceLiteral createXmlSequence(TemplateExpressionNode expressionNode) {
```
Comment (rdhananjaya) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java:4066]: I think we can make this private

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmInstructionGen.java [hunk 7]
```diff
@@ -1696,6 +1697,13 @@ void generateNewXMLProcIns(BIRNonTerminator.NewXMLProcIns newXMLPI) {
         this.storeToVar(newXMLPI.lhsOp.variableDcl);
     }
 
+    void generateXMLSequenceIns(BIRNonTerminator.NewXMLSequence xmlSequenceIns) {
+        this.mv.visitMethodInsn(INVOKESTATIC, XML_FACTORY, "createXmlSequence",
+                String.format("()L%s;", XML_SEQUENCE),
+                false);
```
Comment (rdhananjaya) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmInstructionGen.java:1703]: This argument can goto L:1702

**Review by rdhananjaya — APPROVED** [review]

**Pull request merged**